File: backend/Minigram/Minigram.Core/Utils/Assertions.cs
using System.Diagnostics.CodeAnalysis;

namespace Minigram.Core.Utils
{
    public static class Assertions
    {
        public static void ThrowIfNullOrEmpty([NotNull] Guid? value, string? paramName = null)
        {
            string message = $"{(string.IsNullOrWhiteSpace(paramName) ? nameof(Guid) : paramName)} cannot be null or empty";

            if (value is null || value == Guid.Empty)
            {
                ThrowArgumentException(paramName, message);
            }
        }

        [DoesNotReturn]
        private static void ThrowArgumentException(string? paramName, string message) => throw new ArgumentException(message, paramName);
    }
}

File: backend/python/minigram_core/utils/assertions.py
import uuid


def throw_if_null_or_empty(value: uuid.UUID | None, param_name: str | None = None) -> None:
    name = param_name if param_name else "Guid"
    message = f"{name} cannot be null or empty"

    if value is None or value == uuid.UUID(int=0):
        raise ValueError(message)
